[dhruvir29] iP

dhruvir29 opened the pull request:
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

# Buddy

Buddy frees your mind of having to remember things you need to do. It's,
- text-based
- easy to use
- super fast!

All you need to do is,
1. download it from [here](https://github.com/dhruvir29/ip).
2. add your tasks
3. let it manage your tasks for you 😉

**Features:**
- [x] Adding events, deadlines and todos
- [x] Finding tasks 
- [x] Marking tasks as done

If you Java programmer, you can use it to practice Java too. Here's the main method:
```ruby
 public static void main(String[] args) {
        Ui ui = new Ui();
        TaskList tasks = new TaskList();
        Storage storage = new Storage("data/duke.txt", tasks);
        boolean isRunning = true;
        Parser parser = new Parser();

        try {
            storage.saveTasks();
        } catch (Exception e) {
            e.printStackTrace();
        }

        Scanner scanner = new Scanner(System.in);
        ui.printWelcomeMessage();
        while (isRunning) {
            String input = scanner.nextLine();
            if (input.equals("bye")) {
                ui.printGoodByeMessage();
                break;
            }
            parser.addToList(input, storage, tasks);
        }
    }
``

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

dhruvir29 committed:
greet and exit

dhruvir29 committed:
echo user's input

dhruvir29 committed:
adds user input to a list and displays it when requested

dhruvir29 committed:
mark tasks as done/ change status back

dhruvir29 committed:
ToDos, Events, Deadlines included

dhruvir29 committed:
handling input errors

dhruvir29 committed:
can delete tasks from the list

dhruvir29 committed:
A-TextUiTesting

dhruvir29 committed:
Optimisations done on Week 1 Code

dhruvir29 committed:
Save tasks in hard disk

dhruvir29 committed:
Revert "Save tasks in hard disk"

This reverts commit 33937392362668c29227a72e6fde2d50dc182ef3.

dhruvir29 committed:
save

dhruvir29 committed:
Merge branch 7

dhruvir29 committed:
Save Task

dhruvir29 committed:
Merge branch 'branch-Level-7'

dhruvir29 committed:
First iteration of Save Dates and Times

dhruvir29 committed:
Save Dates and Times

dhruvir29 committed:
Merge branch 'branch-Level-8'

dhruvir29 committed:
Add class Storage

dhruvir29 committed:
Add class TaskList

dhruvir29 committed:
Add class Ui

dhruvir29 committed:
Add Parser class

dhruvir29 committed:
Clean up Parser class

dhruvir29 committed:
Use more Oop

dhruvir29 committed:
Add Packages

dhruvir29 committed:
Add Packages

dhruvir29 committed:
Merge branch 'add-gradle-support'

dhruvir29 committed:
Add JUnit Tests

dhruvir29 committed:
Create Jar file

dhruvir29 committed:
Add JavaDocs

dhruvir29 committed:
Improve the Coding Standard

dhruvir29 committed:
Add improvements to the JavaDocs

dhruvir29 committed:
Merge branch 'branch-A-CodingStandard'

dhruvir29 committed:
Add Find function

dhruvir29 committed:
Merge branch 'Level-9'

pangyyen commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,169 @@
+                }
+
+                if (!input.contains("/by"))  {
+                    throw new DukeException("\t Hey bud! Please include when the deadline is! " +
+                            "\n\t For example you can type: deadline read /by 2023-09-01 1700");
```
Would it be better to break a new line before operator '+'? 

coding standard link for your reference :) 
- https://se-education.org/guides/conventions/java/intermediate.html , 
- session: Layout > "Place line break to improve readability"

pangyyen commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,169 @@
+                    throw new DukeException("\t Sorry! The description of an event cannot be empty :(");
+                }
+                if (!input.contains("/from")) {
+                    throw new DukeException("\t Hey bud! Please include when the event is!" +
+                            "\n\t For example you can type: event hangout /from 2023-09-01 1700 /to 2023-09-01 2000");
```
same as the line breaking suggestion mentioned above

pangyyen commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,169 @@
+                            "\n\t For example you can type: event hangout /from 2023-09-01 1700 /to 2023-09-01 2000");
+                }
+                if (!input.contains("/to")) {
+                    throw new DukeException("\t Hey bud! Please include when the end date of the event is!" +
+                            "\n\t For example you can type: event hangout /from 2023-09-01 1700 /to 2023-09-01 2000");
```
same as the line breaking suggestion mentioned above

pangyyen commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,169 @@
+        if (e instanceof DukeException) {
+            System.out.println(e.getMessage());
+        } else if (e instanceof DateTimeException) {
+            System.out.println("\tPlease put a valid date and time in the format YYYY-MM-DD HHMM." +
+                    "\n\tFor example: 2023-08-08 1800");
```
same as the line breaking suggestion mentioned above

pangyyen reviewed (commented):
other than the minor line breaking suggestion, good job for the overall coding standard!

wr1159 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,64 @@
+import duke.Exception.DukeException;
+import duke.task.ToDo;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Perhaps you want to put line spacing in between the imports from different packages to make it more readable? I notice the same in some other files.

```suggestion
package duke;

import duke.Exception.DukeException;
import duke.task.ToDo;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
```

wr1159 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,64 @@
+        parser.addToList("todo hi", storage, tasks);
+        assertEquals("[T][ ] hi", tasks.getTask(0).toString());
+    }
+
+
+
+
+
+
```
Perhaps all this empty space can be removed to save space?
```suggestion
```

wr1159 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,169 @@
+                }
+                tasks.markTaskAsDone(taskIndex - 1);
+                storage.writeTasksToFile(tasks);
+                ui.printMarkTasksAsDone(taskIndex, tasks);
```
Perhaps there could be more spacing to make this readable? Not sure I like the lines clumped up together but that could be a personal preference too. Similar with the following lines.

```suggestion
                int taskIndex = Integer.parseInt(input.substring(5));
                
                if (tasks.getTask(taskIndex - 1).getStatusIcon() == "X") {
                    throw new DukeException("\tThis task has already been marked as done!");
                }
                
                tasks.markTaskAsDone(taskIndex - 1);
                storage.writeTasksToFile(tasks);
                ui.printMarkTasksAsDone(taskIndex, tasks);
```

wr1159 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,169 @@
+
+    /**
+     * Deals with user inputs beginning with mark, unmark,
+     * delete, list.
```
Was there a reason why delete and list are on another line?

wr1159 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,169 @@
+    }
+
+
+    /**
+     * Deals with user commands to add a task to the tasklist.
+     * If the command is not one that adds a task, it will call the userCommand
+     * method.
+     *
+     * @param input what the user is typing in.
+     * @param storage the storage that is being used.
+     * @param tasks the TaskList that is being used to store the tasks.
+     */
```
I like your JavaDocs comments, they're quite descriptive and helpful to me! 👍

wr1159 reviewed (commented):
Overall LGTM, I left a few comments regarding spacing for your consideration! Good job on your JavaDocs!

dhruvir29 committed:
Add CheckStyle

dhruvir29 committed:
Add the Command classes

dhruvir29 committed:
Set up JavaFX

dhruvir29 committed:
Change to return String

dhruvir29 committed:
Add GUI

dhruvir29 committed:
Changes made to build.gradle file

dhruvir29 committed:
Improve Code Quality

Made the method in parser shorter so that it does not exceed 30 LOC and also
removed redundant code present.

Also added some comments

dhruvir29 committed:
Merge pull request #1 from dhruvir29/branch-A-CodeQuality

Improve Code Quality

dhruvir29 committed:
Add Assertions.

Assertions were added to make sure that the task index was valid, meaning
above 0 and less than the size of the tasklist. Some other assertion also made.

dhruvir29 committed:
Merge branch 'master' into branch-A-Assertions

dhruvir29 committed:
Merge pull request #2 from dhruvir29/branch-A-Assertions

Add Assertions.

dhruvir29 committed:
BCD Extension.

Added C-Help which gives a list of the commands in the chatbot when the user types help.
Also added C-MassOps allowing users to delete more than 1 task at a time.

dhruvir29 committed:
Made some modifications

dhruvir29 committed:
Update README.md

dhruvir29 committed:
Make further changes

dhruvir29 committed:
Add representative screenshot

dhruvir29 committed:
Update README.md

dhruvir29 committed:
Update README.md

dhruvir29 committed:
Edit tasklist test class

dhruvir29 committed:
Make changes and delete dead code

dhruvir29 committed:
Edit comments

dhruvir29 committed:
Update README.md
